### PR TITLE
test: Handle timestamps without microseconds in combine_logs

### PR DESCRIPTION
**Pull request by laanwj**
The log messages before command line arguments have been parsed will not have microseconds. Make parsing robust to lack of microseconds so that sorting is done correctly, and make their timestamps show as '.000000' to line up the log.

Before (note the weird sorting and ordering):
```
 node0 
2018-05-02T12:16:13Z 
 node1 
2018-05-02T12:16:13Z 
 node2 
2018-05-02T12:16:13Z 
 test  2018-05-02T12:16:13.310000Z TestFramework (INFO): Initializing test directory /tmp/testdssm30sn 
```

After:
```
 node0 2018-05-02T12:16:13.000000Z  
 node1 2018-05-02T12:16:13.000000Z  
 node2 2018-05-02T12:16:13.000000Z  
 test  2018-05-02T12:16:13.310000Z TestFramework (INFO): Initializing test directory /tmp/testdssm30sn 
 test  2018-05-02T12:16:13.314000Z TestFramework.node0 (DEBUG): bitcoind started, waiting for RPC to come up 
 test  2018-05-02T12:16:13.316000Z TestFramework.node1 (DEBUG): bitcoind started, waiting for RPC to come up 
 test  2018-05-02T12:16:13.319000Z TestFramework.node2 (DEBUG): bitcoind started, waiting for RPC to come up 
```

### CODE REVIEW

File: test/functional/combine_logs.py [hunk 1]
```diff
@@ -73,10 +73,12 @@ def get_log_events(source, logfile):
                 # if this line has a timestamp, it's the start of a new log event.
                 time_match = TIMESTAMP_PATTERN.match(line)
                 if time_match:
-                    if event:
-                        yield LogEvent(timestamp=timestamp, source=source, event=event.rstrip())
-                    event = line
-                    timestamp = time_match.group()
+                    if event or timestamp:
```
Comment (laanwj) [inline test/functional/combine_logs.py]: Could keep this as `if event` to not show the (initial) empty log message. That's a matter of taste, I guess.

**Comment by MarcoFalke** [comment]
Noting that this is fixed by #13088

**Comment by laanwj** [comment]
Ok...

**Comment by MarcoFalke** [comment]
This is fixed in 

* log: Log early messages #16112 

as a side effect of changing the `     LogPrintf("\n\n\n\n\n");` into a          `FileWriteStr("\n\n\n\n\n", m_fileout);`

**Pull request closed without merging**